/*
 * This file is part of the NUbots Codebase.
 *
 * The NUbots Codebase is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * The NUbots Codebase is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
 *
 * Copyright 2013 NUBots <nubots@nubots.net>
 */

package messages.support.nubugger.proto;

import "messages/Vector.proto";

message DrawObject {
    enum Shape {
        ARROW = 1;
        BOX = 2;
        CIRCLE = 3;
        CYLINDER = 4;
        POLYLINE = 5;
        PYRAMID = 6;
        RECTANGLE = 7;
        SPHERE = 8;
    }

    required Shape shape = 1;
    optional string name = 2;

    required vec3 position = 3;
    optional vec3 direction = 4;
    optional vec3 target = 5;
    optional float width = 6;
    optional float height = 7;
    optional vec3 rotation = 8;
    optional vec3 color = 9; // rgb
    optional float radius = 10;
    optional float top_radius = 11;
    optional float bottom_radius = 12;
    repeated vec3 vertices = 13;
    optional uint32 faces = 14;
    optional float lineWidth = 15;
    optional float length = 16;
    optional float depth = 17;
    optional bool fill = 18;

    optional float timeOut = 19;
}

message DrawObjects {
    repeated DrawObject objects = 1;
}